sunpy/image/coalignment.py

```diff
@@ -64,9 +66,16 @@ def calculate_shift(this_layer, template):
         Pixel shifts ``(yshift, xshift)`` relative to the offset of the template
         to the input array.
     """
-    # Repair any NANs, Infs, etc in the layer and the template
-    this_layer = repair_image_nonfinite(this_layer)
-    template = repair_image_nonfinite(template)
+    if repair_nonfinite:
+        # Repair any NANs, Infs, etc in the layer and the template
+        # This behaviour is deprecated
+        warnings.warn('The value True for the repair_nonfinite kwarg is deprecated',
```

Comment:
Is the plan to leave this kwarg in and just default it to `False` post 2.0, or remove it altogether.

From the wording of this warning it sounds like the former, but then you are deprecating the repair function, so I am a little confused.

Reply:
So I'm still fuzzy on the policy here, but I think your comment from above makes the most sense:

> I think what we could do is add the kwarg and add a deprecation warning (for the behaviour change and kwarg removal) now, leave that in place until 2.0, then change the behaviour for 2.1 and remove it in 3.0.

So `repair_nonfinite` will default to `True` for now, will default to `False` starting at 2.1, and then be removed, along with the `repair_image_nonfinite` function, in 3.0

Reply:
OK, makes sense. I feel this message should be changed to make it clear the behavior of repairing is deprecated not just the True value.

Reply:
Done!